cranelift-codegen/src/ir/types.rs

```diff
@@ -106,8 +106,8 @@ impl Type {
         self.replace_lanes(match self.lane_type() {
             B8 | I8 => B8,
             B16 | I16 => B16,
-            B32 | I32 | F32 => B32,
-            B64 | I64 | F64 => B64,
+            B32 | I32 | F32 | R32 => B32,
+            B64 | I64 | F64 | R64 => B64,
```

Comment:
We should probably assert there aren't ref types in here, instead (otherwise, this means we're trying to resolve operand constraints for a derived type variable, which base contains ref types, and derivation is as_bool: we're trying to convert ref types to bools!).